tests/python/unit/test_base.py

```diff
@@ -0,0 +1,68 @@
+import pandas as pd
+import os.path
+
+from data_loader import DataLoader
+from tempfile import NamedTemporaryFile
+from nearest_neighbor.base import NearestNeighborSearch
+
+
+def test_build_save_index():
+
+    params = {'unique_id_column': 'images',
+              'feature_columns': ['prediction'],
+              'algorithm': 'annoy',
+              'expert': True,
+              'annoy_metric': 'angular',
+              'annoy_num_trees': 10}
+
+    # Load data into vector format for indexing
+    columns = [params["unique_id_column"]] + params["feature_columns"]
+    input_df = pd.read_csv('./tests/resources/caltech_embeddings.csv')
+    # Restrict to selected columns
+    input_df = input_df[columns]
+    data_loader = DataLoader(params["unique_id_column"], params["feature_columns"])
+    (vector_ids, vectors) = data_loader.convert_df_to_vectors(input_df)
+    nearest_neighbor = NearestNeighborSearch(num_dimensions=vectors.shape[1], **params)
+    tmp = NamedTemporaryFile()
+    nearest_neighbor.build_save_index(vectors=vectors, index_path=tmp.name)
+    file_exist = os.path.isfile(tmp.name)
+    tmp.close()
+    # Test if file is created as a result
+    assert file_exist
```

Comment:
Maybe better to `assert` right after `file_exist`? Or just `assert os.path.isfile(tmp.name)` directly?

Reply:
Done